prototypes/javaspec-engine/src/main/java/info/javaspec/engine/SpecClassDescriptor.java

```diff
@@ -34,21 +36,42 @@ public Type getType() {
 
 	/* JavaSpec */
 
+	// Entry point into the discovery process
 	public void discover() {
+		setCurrentContainer(this);
 		this.declaringInstance.declareSpecs(this);
 	}
 
+	@Override
+	public void describe(String what, BehaviorDeclaration declaration) {
+		TestDescriptor container = currentContainer();
+		DescribeDescriptor child = DescribeDescriptor.about(container.getUniqueId(), what);
+		container.addChild(child);
+
+		setCurrentContainer(child);
+		declaration.declare();
+		setCurrentContainer(this);
```

Comment:
This is what would keep us from declaring more specs after a nested `describe`.  Those specs would wind up on the container for the `SpecClass` instead.

Reply:
...and this is saved for future work, to keep this PR focused on a small, but significant feature.